### PR TITLE
main: add decancer

### PR DESCRIPTION
**Pull request by vierofernando**
originally gonna add `isEmpty` but i think this is more epic

c:

### CODE REVIEW

**Comment by VoltrexMaster** [comment]
May you change the double quotes in the constants to single quotes for consistency?

**Comment by vierofernando** [comment]
okie! got ya covered! <3

File: index.js [hunk 1]
```diff
@@ -194,6 +246,28 @@ class Functions {
     return string.length > length ? string.slice(0, length) + (placeholder ? placeholder.toString() : '...') : string;
   }
 
+  /**
+   * Decancers a string. Reduces everything from zalgos, fancy characters, etc. to alphanumeric characters.
+   * NOTE: The string returned will ALWAYS be in lowercase.
+   * @param {string} text The cancerous string to decancer.
+   * @returns {string} The cleaned string. This may break certain things like special characters. Only use for certain purposes like filtering/censoring.
+   */
+  decancer(text) => {
```
Comment (VoltrexMaster) [inline index.js:255]:
```suggestion
  decancer(text) {
```
Unnecessary arrow expression.

File: index.js [hunk 2]
```diff
@@ -194,6 +246,28 @@ class Functions {
...
+    if (typeof text !== 'string') throw new TypeError('First parameter must be a type of string');
+    
+    text = text
+      .toLowerCase()
+      .replace(cleanerRegex, "");
```
Comment (VoltrexMaster) [inline index.js:260]:
```suggestion
      .replace(cleanerRegex, '');
```
Missed this here.

Comment (vierofernando) [reply index.js:260]: oops, forgot to change that one, the code was copied off of my gist, brb

**Review by VoltrexMaster — APPROVED** [review]
❤️

**Review by bl-ue — APPROVED** [review]
Wow, this is really cool!  @vierofernando, if I may ask, where did you find this code?

**Comment by vierofernando** [comment]
> Wow, this is really cool! @vierofernando, if I may ask, where did you find this code?
@bl-ue  i didn't found this anywhere, i made them from scratch!

i sampled most of the fancy texts from https://lingojam.com/FancyTextGenerator and https://www.fancytextguru.com/,
the zalgo characters from this file: https://github.com/michaelrhodes/to-zalgo/blob/master/lib/chars.js,
and others by manually searching through https://www.codetable.net/groups

**Comment by bl-ue** [comment]
Wow, that's really cool!  I'm going to at least bookmark this PR for future reference :)

**Comment by vierofernando** [comment]
> Wow, that's really cool! I'm going to at least bookmark this PR for future reference :)

thanks c: @bl-ue

**Pull request merged**